### ISSUE TITLE
Suggestion: Inline pipeline for iterating pipes

### ISSUE THREAD

**Issue by Laurens-makel** [frankframework/frankframework#6572]
**❗ please do not add sensitive information in issues, you can provide extra information via email using issue number as reference ❗**

**Describe the issue**
The most convenient way for iterative processing is to use the different types of iterating pipes, with the `ForEachChildElementPipe` being the most commonly used.

These pipes accept a sender to send their messages to, in a lot of cases there is no need to send iterative messages to an external system but do some iterative internal processing instead.

Currently you could chain senders using the `SenderSeriesSender` to execute some logic for each iteration, however in certain cases you'd need to write to session and that's not possible using the described setup. Usually this is solved by creating so called sub adapters.

An alternative approach could be to allow inline `<pipeline>` elements instead. This nested pipeline basically inherits all values from the current session and on session end writes the modifications back into the main session (could be problematic with parallel processing).

Besides a simpler syntax, the main differences in this approach is isolation. The nested pipeline is not an adapter thus cannot be called by other adapters except the one that declares it and does not show up as an adapter in the web console. 

Please let me know what you think :)!

**Comment by nielsm5** [comment]
The SenderWrapperSender allows you to read/write to the session

**Comment by Laurens-makel** [comment]
> The SenderWrapperSender allows you to read/write to the session

Thanks, this seems to be the missing component that I was looking for :)!

My initial reason to draft this proposal might be 'solved' with this answer however there are still some things which are not really possible in iterative processing without using sub-adapters:

- `SenderSeriesSener` and `SenderWrapperSender` only allow sender elements, pipes such as `XmlIfPipe` cannot be used so (partially) conditional processing is not possible (execute step ABC, conditionally DEF if a certain condition is met). Base64Pipe cannot be used, it's not possible to decode/encode data on a single iteration. PutInSessionPipe cannot be used to put multiple values in session. CsvParserPipe cannot be used to parse a CSV on a single iteration... etc etc etc 

- Only technical exceptions can be handled with a single exception forward, it's not possible to return a functioncal exception if a certain condition is not met in an iteration.'

**Comment by nielsm5** [comment]
Yup I understand, was more to give you a quick pointer / some help to continue. But I agree this is less optimal...
The problem is that it would complicate things a bit too much, the whole idea between pipes and senders is that senders are not aware of their exits, they just do 'something' with a success or exception. Pipes on the other hand are aware of their exits.

**Comment by jacodg** [comment]
We hope #3270 will make this issue obsolete
